homeassistant/components/bring/__init__.py

```diff
@@ -0,0 +1,75 @@
+"""The Bring! integration."""
+from __future__ import annotations
+
+import logging
+
+from python_bring_api.bring import Bring
+from python_bring_api.exceptions import (
+    BringAuthException,
+    BringParseException,
+    BringRequestException,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import BringDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.TODO]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Bring! from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    email = entry.data[CONF_EMAIL]
+    password = entry.data[CONF_PASSWORD]
+
+    bring = Bring(email, password)
+
+    try:
+        await hass.async_add_executor_job(bring.login)
+        await hass.async_add_executor_job(bring.loadLists)
+    except BringRequestException as e:
+        raise ConfigEntryNotReady(
+            f"Timeout while connecting for email '{email}'"
+        ) from e
+    except BringAuthException as e:
+        _LOGGER.error(
+            "Authentication failed for '%s', check your email and password",
+            email,
+        )
+        raise ConfigEntryError(
+            f"Authentication failed for '{email}', check your email and password"
+        ) from e
+    except BringParseException as e:
+        _LOGGER.error(
+            "Failed to parse request '%s', check your email and password",
+            email,
+        )
+        raise ConfigEntryError(
+            f"Failed to parse request '{email}', check your email and password"
+        ) from e
```

Comment:
Why do we expect a non parseable response is an error?

Reply:
What else should it be if I cannot parse the response json from the api/server?

Reply:
I mean, Imagine the service returning a 503 because someone messed up a server config. This would be something that could be fixed in an hour, thus I think retrying is something we should do.

Reply:
Ok, I can change it so the `JSONParse` errors are rather retried instead of aborted.